src/Dumplie/Application/Query/Customer/Result/Cart.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Query\Customer\Result;
+
+use Dumplie\Application\Exception\InvalidArgumentException;
+
+final class Cart
+{
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    /**
+     * @var float
+     */
+    private $totalPrice;
+
+    /**
+     * @param array|CartItem[] $items
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $items = [])
+    {
+        $this->totalPrice = 0;
+
+        foreach ($items as $item) {
+            if (!$item instanceof CartItem) {
+                throw new InvalidArgumentException();
```

Comment:
Also, in one of my playground projects I started to use static constructors for exceptions that require set of arguments used in predefined message.
```php
class InvalidArgumetnException
{
	public static function invalidItemClass($item): InvalidArgumentException
	{
		return new static('Item must be an instance of %s, got %s', CartItem::class, gettype($item));
	}
}

throw InvalidArgumentException::invalidItemClass($item);
```